public/js/reducers/sources.js

```diff
@@ -156,24 +171,30 @@ function updateTabList(state, source, tabIndex) {
  * @memberof reducers/sources
  * @static
  */
-function getNewSelectedSourceId(state, id) : ?Source {
+function getNewSelectedSourceId(state, id: any) : ?Source {
```

Comment:
Hmm, hope it's alright, but I'm taking a stab at some of the unit tests + `getNewSelectedSourceId` logic. This function is **complex**...

My thinking is that:
a) if the current active tab is still around, it should remain active
b) if it's nuked, the next available tab to its left should be activated. 

This logic could simplify things quite a bit as well

Also, I hope you enjoy, I was curious so i took a look at how atom handles tabs and found this [view](https://github.com/atom/tabs/blob/master/lib/tab-bar-view.coffee). This was the first time I inspected atom for a profit... was pretty neat :)